File: Security project/common/src/main/java/sni/common/models/dtos/FileLogDTO.java
package sni.common.models.dtos;

import lombok.Data;
import sni.common.models.enums.Operation;

import java.time.LocalDateTime;

@Data
public class FileLogDTO
{
    private Long logId;
    private String description;
    private LocalDateTime timestamp;
    private Operation operation;
    private FileBasicDTO affectedFile;
    private UserBasicDTO createdBy;
}

File: Files frontend/src/app/models/FileLogDTO.ts
import { FileBasicDTO } from "./FileBasicDTO";
import { Operation } from "./Operation";
import { UserBasicDTO } from "./UserBasicDTO";

export interface FileLogDTO{
    logId: number;
    description: string;
    timestamp: Date;
    operation: Operation;
    affectedFile: FileBasicDTO;
    createdBy: UserBasicDTO;
}